components/content_settings/core/browser/ad_block_only_mode_content_settings_utils_unittest.cc

```diff
@@ -0,0 +1,209 @@
+#include "components/content_settings/core/common/content_settings_enums.mojom.h"
+#include "components/content_settings/core/common/content_settings_metadata.h"
+#include "components/content_settings/core/common/content_settings_pattern.h"
+#include "components/content_settings/core/common/content_settings_types.h"
+#include "components/content_settings/core/common/content_settings_utils.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+namespace content_settings {
+
+class AdBlockOnlyModeContentSettingsUtilsTest : public testing::Test {
+ public:
+  AdBlockOnlyModeContentSettingsUtilsTest() = default;
+  ~AdBlockOnlyModeContentSettingsUtilsTest() override = default;
+
+  void CheckMetadata(const RuleMetaData& metadata) {
+    EXPECT_EQ(metadata.session_model(), mojom::SessionModel::DURABLE);
+    EXPECT_EQ(metadata.expiration(), base::Time());
+    EXPECT_EQ(metadata.lifetime(), base::TimeDelta());
+  }
+};
+
+TEST_F(AdBlockOnlyModeContentSettingsUtilsTest,
+       IsAdBlockOnlyModeContentSettingsType) {
+  EXPECT_TRUE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::JAVASCRIPT, /*is_off_the_record*/ false));
+  EXPECT_TRUE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::COOKIES, /*is_off_the_record*/ false));
+  EXPECT_TRUE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::BRAVE_COOKIES, /*is_off_the_record*/ false));
+  EXPECT_TRUE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::BRAVE_REFERRERS, /*is_off_the_record*/ false));
+  EXPECT_TRUE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::BRAVE_ADS, /*is_off_the_record*/ false));
+  EXPECT_TRUE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::BRAVE_TRACKERS, /*is_off_the_record*/ false));
+  EXPECT_TRUE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::BRAVE_COSMETIC_FILTERING,
+      /*is_off_the_record*/ false));
+  EXPECT_TRUE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::BRAVE_FINGERPRINTING_V2,
+      /*is_off_the_record*/ false));
+  EXPECT_TRUE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::BRAVE_REMEMBER_1P_STORAGE,
+      /*is_off_the_record*/ false));
+  EXPECT_TRUE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::BRAVE_HTTPS_UPGRADE, /*is_off_the_record*/ false));
+
+  EXPECT_FALSE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::GEOLOCATION, /*is_off_the_record*/ false));
+  EXPECT_FALSE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::NOTIFICATIONS, /*is_off_the_record*/ false));
+  EXPECT_FALSE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::IMAGES, /*is_off_the_record*/ false));
+  EXPECT_FALSE(IsAdBlockOnlyModeContentSettingsType(
+      ContentSettingsType::BRAVE_SHIELDS, /*is_off_the_record*/ false));
+}
+
+TEST_F(AdBlockOnlyModeContentSettingsUtilsTest,
+       IsAdBlockOnlyModeContentSettingsType_OffTheRecord) {
```

Comment:
Do we need the _? Maybe IsAdBlockOnlyModeContentSettingsTypeWhenOffTheRecord

Reply:
Changed